File: algorithm/sort.go
package main

import (
	"fmt"
)

func main() {
	data := []int{74, 59, 238, -784, 9845, 959, 905, 0, 0, 42, 7586, -5467984, 7586}
	fmt.Println(data)
	BubbleSort(data)
	fmt.Println(data)
	data = []int{74, 59, 238, -784, 9845, 959, 905, 0, 0, 42, 7586, -5467984, 7586}
	fmt.Println(data)
	InsertSort(data)
	fmt.Println(data)
}

func BubbleSort(a []int) {
	for i := 0; i < len(a)-1; i++ {
		for j := 0; j < len(a)-i-1; j++ {
			if a[j] > a[j+1] {
				a[j], a[j+1] = a[j+1], a[j]
			}
		}
	}
}

func InsertSort(a []int) {
	for i := 1; i < len(a); i++ {
		for j := i; j > 0; j-- {
			if a[j] < a[j-1] {
				a[j], a[j-1] = a[j-1], a[j]
			} else {
				break
			}
		}
	}
}

File: algorithm/sort.py
def BubbleSort(Iterator):
    """ 冒泡排序 """
    a = list(Iterator)
    for i in range(len(a)):
        for j in range(len(a)-1-i):
            if a[j] > a[j+1]:
                a[j], a[j+1] = a[j+1], a[j]
    return a

def InsertSort(Iterator):
    """ 插入排序 """
    a = list(Iterator)
    for i in range(1, len(a)):
        for j in range(i, 0, -1):
            if a[j] < a[j-1]:
                a[j], a[j-1] = a[j-1], a[j]
    return a

def main():
    data = [74, 59, 238, -784, 9845, 959, 905, 0, 0, 42, 7586, -5467984, 7586]
    print(BubbleSort(data))
    print(InsertSort(data))

if __name__ == '__main__':
    main()